### PR TITLE
Add API functionality

### PR DESCRIPTION
**Pull request by george-bitdotio**
**Summary**
- Basic project setup:
  - Build setup with `tsc`
  - Testing setup with Jest
  - ESLint
  - Prettier
  - Run tests in CI against Node 16 & 18
- Library structure:
  - Like `python-bitdotio`, lib is structured around an `SDK` object which is instantiated by passing an API key to the `bitdotio` function
  - Submodules include `apiClient`, `errors`, and `utils`, which pretty much consist of what they sound like they consist of
  - Typings are provided for all API methods
- This PR reaches feature-parity with `python-bitdotio` for API methods:
  - `listDatabases`
  - `createDatabase`
  - `getDatabase`
  - `updateDatabase`
  - `deleteDatabase`
  - `createImportJob`
  - `getImportJob`
  - `createExportJob`
  - `getExportJob`

**Testing Strategy**
Unit tests added for all utils, SDK methods, and API client methods. Also tested everything e2e

### CODE REVIEW

File: lib/apiClient.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    method: string,
+    path: string,
+    init?: RequestInit,
+  ): Promise<T> {
+    const response = await fetch(this._url(path), {
+      ...init,
+      headers: {
+        ...init?.headers,
+        ...this._headers,
+      },
```
Comment (eric-bit-io) [inline lib/apiClient.ts:36]: maybe doesn't matter, but should User-Agent header be overrideable?

**Review by jmo-qap — COMMENTED** [review]
Structure LGTM although I don't know TS at all. Also configured all the branch protection rules. Thanks for putting in tests and auto-running them!

File: package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-prettier": "^4.2.1",
+    "eslint-plugin-unused-imports": "^2.0.0",
+    "jest": "^29.3.1",
+    "prettier": "^2.8.0",
+    "ts-jest": "^29.0.3",
+    "typescript": "^4.9.3"
+  },
+  "dependencies": {
+    "node-fetch": "^2.6.7"
```
[the remark this answers is not in the document]
Comment (george-bitdotio) [reply package.json:37]: No, this lib uses a lot of _very_ node-specific APIs

File: lib/apiClient.ts (hunk 1 above)
Comment (george-bitdotio) [reply lib/apiClient.ts:36]: No I think it's useful for our analytics purposes to force the UA here

File: test/apiMethods.test.ts [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+      .spyOn(nf, "default")
+      .mockResolvedValueOnce(mockResponse(200, { databases: expected }));
+    const result = await b.listDatabases();
+    expect(result).toEqual(expected);
+  });
+  test("listDatabases error", async () => {
+    const status = 400;
+    const data = { error: "whoops" };
+    jest.spyOn(nf, "default").mockResolvedValueOnce(mockResponse(status, data));
+    try {
```
Comment (eric-bit-io) [inline test/apiMethods.test.ts:32]: All of these async thrown error tests look odd to me. I think the idiomatic approach is using rejects: https://jestjs.io/docs/expect#rejects

something like 
```
test('rejects to octopus', () => {
  // make sure to add a return statement
  return expect(Promise.reject({name: 'octopus', status: 401})).rejects.toEqual(
    expect.objectContaining({name: 'octopus', status:401}) );
});
```

**Review by eric-bit-io — APPROVED** [review]
Mostly looks great. Non blocking question about user again and non-blocking style feedback on async failure tests

File: test/apiMethods.test.ts (hunk 3 above)
Comment (george-bitdotio) [reply test/apiMethods.test.ts:32]: You're right. I didn't realize I could do:
```js
await expect(b.someMethod()).rejects.toThrow(new ApiError(...));
```
will fix 🤦‍♀️

**Pull request merged**